### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by dream-x** [deanperry/shoppe-paypal#1]
After add Gemfile - gem "shoppe-paypal" catch this error. Ruby 2.1.2, Rails 4.0.1, shoppe-1.0.5

```
/Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/shoppe-1.0.2/lib/shoppe/settings.rb:16:in `method_missing': undefined method `keys' for "translation missing: en.shoppe.settings.defaults":String (NoMethodError)
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/bundler/gems/shoppe-paypal-bd883d8d5c84/lib/shoppe/paypal.rb:36:in `setup'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/bundler/gems/shoppe-paypal-bd883d8d5c84/lib/shoppe/paypal/engine.rb:6:in `block in <class:Engine>'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/initializable.rb:30:in `instance_exec'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/initializable.rb:30:in `run'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/initializable.rb:55:in `block in run_initializers'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:347:in `block in each_strongly_connected_component'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `call'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:224:in `tsort_each'
    from /Users/kmax/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/initializable.rb:54:in `run_initializers'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/application.rb:215:in `initialize!'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/railtie/configurable.rb:30:in `method_missing'
    from /Users/kmax/P/_/3d-humanity/config/environment.rb:5:in `<top (required)>'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/activesupport-4.0.1/lib/active_support/dependencies.rb:229:in `require'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/activesupport-4.0.1/lib/active_support/dependencies.rb:229:in `block in require'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/activesupport-4.0.1/lib/active_support/dependencies.rb:214:in `load_dependency'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/activesupport-4.0.1/lib/active_support/dependencies.rb:229:in `require'
    from /Users/kmax/P/_/3d-humanity/config.ru:3:in `block in <main>'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/builder.rb:55:in `instance_eval'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/builder.rb:55:in `initialize'
    from /Users/kmax/P/_/3d-humanity/config.ru:in `new'
    from /Users/kmax/P/_/3d-humanity/config.ru:in `<main>'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/builder.rb:49:in `eval'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/builder.rb:49:in `new_from_string'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/builder.rb:40:in `parse_file'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/server.rb:277:in `build_app_and_options_from_config'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/server.rb:199:in `app'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/commands/server.rb:48:in `app'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/rack-1.5.2/lib/rack/server.rb:314:in `wrapped_app'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/commands/server.rb:75:in `start'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/commands.rb:76:in `block in <top (required)>'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/commands.rb:71:in `tap'
    from /Users/kmax/.rvm/gems/ruby-2.1.2@humanity/gems/railties-4.0.1/lib/rails/commands.rb:71:in `<top (required)>'
    from bin/rails:4:in `require'
    from bin/rails:4:in `<main>'
```

**Comment by deanpcmad** [comment]
It shows from the first line that you are using Shoppe version 1.0.2. Can you `bundle update shoppe` and see if the latest version fixes it? I've not seen this issue before. Is this a new Rails app or an existing one?

**Comment by deanpcmad** [comment]
You should also be able to use the `shoppe-paypal` which is on Rubygems instead of the Github version

**Comment by alexwiththescar** [comment]
Hi, I am getting the same issues. Latest version of Shoppe, Shoppe-paypal, Ruby 2.2.1 and Rails 4.2.

Cant figure the cause, but has somthing to do with Calling the `Shoppe.settings.paypal_` for the API keys I think.

Any advice welcome

Thanks

```
.rvm/gems/ruby-2.2.1/bundler/gems/shoppe-1.0.5/lib/shoppe/settings.rb:16:in `method_missing': undefined method `keys' for "translation missing: en.shoppe.settings.defaults":String (NoMethodError)
.rvm/gems/ruby-2.2.1/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal.rb:36:in `setup'
.rvm/gems/ruby-2.2.1/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal/engine.rb:6:in `block in <class:Engine>'
    from /home/alex/.rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:30:in `instance_exec'
.rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:30:in `run'
.rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:55:in `block in run_initializers'
```

**Comment by deanpcmad** [comment]
Ok I'll see if I can take a look at this over the weekend

On Friday, April 24, 2015, alexwiththescar notifications@github.com wrote:

> Hi, I am getting the same issues. Latest version of Shoppe, Shoppe-paypal,
> Ruby 2.2.1 and Rails 4.2.
> 
> Cant figure the cause, but has somthing to do with Calling the Shoppe
> Setting for the API keys I think.
> 
> Any advice welcome
> 
> Thanks
> 
> .rvm/gems/ruby-2.2.1/bundler/gems/shoppe-1.0.5/lib/shoppe/settings.rb:16:in `method_missing': undefined method`keys' for "translation missing: en.shoppe.settings.defaults":String (NoMethodError)
> .rvm/gems/ruby-2.2.1/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal.rb:36:in `setup'
> .rvm/gems/ruby-2.2.1/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal/engine.rb:6:in`block in class:Engine'
>     from /home/alex/.rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:30:in `instance_exec'
> .rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:30:in`run'
> .rvm/gems/ruby-2.2.1/gems/railties-4.2.0/lib/rails/initializable.rb:55:in `block in run_initializers'
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/deanperry/shoppe-paypal/issues/1#issuecomment-95856257
> .

## 

Sent from Gmail Mobile

**Comment by alexwiththescar** [comment]
Still not too sure what is causing this Issue, but If it of any help, I have found if you can set paypal_client_id and paypal_secret_id in the DB it will work. I removed the gem, opened console ran 

```
paypal_hash = {"paypal_client_id" => ENV['PAYPAL_CLIENT'], "paypal_secret_id" => ENV['PAYPAL_SECRET']}
Shoppe::Setting.update_from_hash(paypal_hash)
```

It now seems to work.

A real fix would be really appreciated! And thanks for all your hard work actually, very gem!

**Comment by deanpcmad** [comment]
Ok thanks, someone emailed me with the same issue I think. Will let you
know :)

On Friday, April 24, 2015, alexwiththescar notifications@github.com wrote:

> Still not too sure what is causing this Issue, but If it of any help, I
> have found if you can set paypal_client_id and paypal_secret_id in the DB
> it will work. I removed the gem, opened console ran
> 
> paypal_hash = {"paypal_client_id" => ENV['PAYPAL_CLIENT'], "paypal_secret_id" => ENV['PAYPAL_SECRET']}
> Shoppe::Setting.update_from_hash(paypal_hash)
> 
> It now seems to work.
> 
> A real fix would be really appreciated! And thanks for all your hard work
> actually, very gem!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/deanperry/shoppe-paypal/issues/1#issuecomment-95925692
> .

## 

Sent from Gmail Mobile

**Comment by deanpcmad** [comment]
I can replicate this error but I can't seem to find why it's happening... Maybe @adamcooke could help? :)

**Comment by 800a7b32** [comment]
+1 for this error. Running Shoppe 1.0.5. I've got the shoppe-stripe gem working fine, but get the same error as listed above when trying to add shoppe-paypal gem

**Comment by deanpcmad** [comment]
Yep I can't seem to find why it's happening. I'll see if I can take a further look tomorrow

**Comment by deanpcmad** [comment]
This is now fixed in v1.1.1 :+1:

**Comment by deanpcmad** [comment]
I have also updated the documentation for a change in the controller code required - http://tryshoppe.com/docs/payment-gateways/paypal
